lane.py

```diff
@@ -387,897 +372,796 @@ def has_next_page(self):
+    )
 
-    def works(self, facets=None, pagination=None):
-        """Find Works that will go together in this Lane.
+    # identifier is a name for this lane that is unique across the
+    # library.  "Adult Fiction" is a good example. We can't have an
+    # adult fiction lane called "Fiction" and a YA fiction lane called
+    # "Fiction".
+    identifier = Column(Unicode)
+
+    # display_name is the name of the lane as shown to patrons.  It's
+    # okay for this to be duplicated across the library, but it's not
+    # okay to have two lanes with the same parent and the same display
+    # name -- that would be confusing.
+    display_name = Column(Unicode)
+
+    # True = Fiction only
+    # False = Nonfiction only
+    # null = Both fiction and nonfiction
+    #
+    # This may interact with lane_genres, for genres such as Humor
+    # which can apply to either fiction or nonfiction.
+    fiction = Column(Boolean, index=True, nullable=True)
+
+    # A lane may be restricted to works classified for specific audiences
+    # (e.g. only Young Adult works).
+    _audiences = Column(ARRAY(Unicode), name='audiences')
+
+    # A lane may further be restricted to works classified as suitable
+    # for a specific age range.
+    _target_age = Column(INT4RANGE, name="target_age", index=True)
+
+    # A lane may be restricted to works available in certain languages.
+    languages = Column(ARRAY(Unicode))
+
+    # A lane may be restricted to works in certain media (e.g. only
+    # audiobooks).
+    media = Column(ARRAY(Unicode))
+
+    # TODO: At some point it may be possible to restrict a lane to certain
+    # formats (e.g. only electronic materials or only codices).
+
+    # Only books licensed through this DataSource will be shown.
+    license_datasource_id = Column(
+        Integer, ForeignKey('datasources.id'), index=True,
+        nullable=True
+    )
 
-        Works will:
+    # Only books on one or more CustomLists obtained from this
+    # DataSource will be shown.
+    _list_datasource_id = Column(
+        Integer, ForeignKey('datasources.id'), index=True,
+        nullable=True
+    )
 
-        * Be in one of the languages listed in `languages`,
-          and not one of the languages listed in `exclude_languages`.
+    # Only the books on these specific CustomLists will be shown.
+    customlists = relationship(
+        "CustomList", secondary=lambda: lanes_customlists,
+        backref="lane"
+    )
 
-        * Be filed under of the genres listed in `self.genre_ids` (or, if
-          `self.include_subgenres` is True, any of those genres'
-          subgenres).
+    # This has no effect unless list_datasource_id or
+    # list_identifier_id is also set. If this is set, then a book will
+    # only be shown if it was seen on an appropriate list within this
+    # number of days. If the number is zero, the book must be
+    # _currently_ on an appropriate list.
+    list_seen_in_previous_days = Column(Integer, nullable=True)
```

Comment:
I don't understand what it means to be "seen" on a list here.

Reply:
This is specific to CustomLists that monitor lists from external sources such as the NYT best-seller list. When a book leaves a best-seller list it's not removed from the corresponding CustomList. Instead, we stop updating its `.most_recent_appearance`. `list_seen_in_previous_days` lets us create a 'best-sellers' lane containing books that are _currently_ on the NYT best-seller list as well as books that were _recently_ on the best-seller list.

I've rewritten the comment in a way that hopefully clarifies things.